### PR TITLE
Adjust Creator Settings Form Width

### PR DESCRIPTION
**Pull request by juliannatetreault**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR adjusts the width of the "Community Name" field, the "Finish" button, and the "Brand Color
 field on the Creator Settings form so that both fields are now the same size as the color selector field, creating more of a "block" feel to the form.

For more information on why this change is being made, please refer to the design critique in the issue referenced below.

## Related Tickets & Documents
Closes issue #15524 

#### Please note that you will need to essentially clear out your database to take this feature for a spin. 🏎️  

To do so, you can either drop your entire database via `rails:db drop` 

_**or**_

You can open a rails console via `rails c` and do the following:
 ```
# Enable the feature flag
FeatureFlag.enable(:creator_onboarding)

# Delete all users 
User.destroy_all

# Set `waiting_on_first_user` to be true
Settings::General.waiting_on_first_user = true

# If you've logged in before set the mascot id to nil
 Settings::General.mascot_user_id = nil
```

Now, run your server and navigate to `http://localhost:3000/` and complete the "Create Account" form. After completing the "Create Account" form, you should be brought to the new Creator Setup page, `${baseUrl}/admin/creator_settings/new?referrer=${baseUrl}`.

Once on the Creator Settings page, you will want to test the changes outlined below:
#### Before:
Screen Shot 2021-12-10 at 6 16 34 AM

#### After:
Screen Shot 2021-12-10 at 6 11 22 AM

#### After (Mobile View):
Screen Shot 2021-12-10 at 6 17 38 AM

### UI accessibility concerns?
N/A - this just changes the width of the form buttons to match the width of the other form fields, like the color selector.

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: there are already tests in place for the Creator Onboarding flow.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
N/A

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by juliannatetreault** [comment]
Rather than reducing the width of the "Community Name" field and "Finish" button, I adjusted the width of the "Community Name" field, "Finish" button, and "Brand Color" field to all be the same. The reason being that when I changed the width of just the "Community Name" field and the "Finish" button to match that of the "Brand Color" field, not only did the form look off in general, but it also looked quite different from the design shown in the design feedback linked in the issue in the PR description and to the original Creator Onboarding design RFC.

For reference, this is what the form looked like when I reduced the width of the "Community Name" field and "Finish" button to match the "Brand Color" field:
Screen Shot 2021-12-10 at 6 24 06 AM

This all being said, I'm happy to adjust the form to the above design if folks decide they prefer it, but personally thought that the current form layout looked more like a "block" (the ultimate goal of this change).

cc @Ridhwana @anujbhavsar96 @nickytonline @JennieOcken

File: app/views/admin/creator_settings/_form.html.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <span class="crayons-field__required crayons-tooltip" data-tooltip="This will set the primary name for your Forem" aria-describedby="community-name-subtitle"></span>
     <p id="community-name-subtitle" class="crayons-field__description">Used as the primary name for your Forem.</p>
   <% end %>
-  <%= text_field_tag :community_name, "", placeholder: "Climbing Life", class: "crayons-textfield", required: true %>
+  <%= text_field_tag :community_name, "", placeholder: "Climbing Life", class: "crayons-textfield flex w-100 m:w-100", required: true %>
```
Comment (nickytonline) [inline app/views/admin/creator_settings/_form.html.erb:7]: Refactor (blocking): No need for any width CSS utility classes or the `flex` CSS utility class. The parent, `.crayons-field` has its display set to flex with a flex-flow of column.

image

```suggestion
  <%= text_field_tag :community_name, "", placeholder: "Climbing Life", class: "crayons-textfield", required: true %>
```

File: app/views/admin/creator_settings/_form.html.erb [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <p id="color-selector-subtitle" class="crayons-field__description">This will be the "accent" color used for buttons, links, etc.</p>
   <% end %>
 
-  <div class="flex w-100 m:w-50 crayons-field">
+  <div class="flex w-100 m:w-100 crayons-field">
```
Comment (nickytonline) [inline app/views/admin/creator_settings/_form.html.erb:19]: Refactor (blocking): No need for width CSS utility classes as the `flex` CSS utility class will take the entire space available and adjust as the page resizes. Also, the `.crayons-field` already sets the `display` to `flex`.

image

```suggestion
  <div class="crayons-field">
```

File: app/views/admin/creator_settings/new.html.erb [hunk 3]
```diff
@@ -37,7 +37,7 @@
       <br>
       <%= render "form" %>
       <div class="crayons-field mt-6 align-left">
-        <%= submit_tag "Finish", class: "crayons-btn btn--primary" %>
+        <%= submit_tag "Finish", class: "crayons-btn btn--primary w-100 m:w-100" %>
```
Comment (nickytonline) [inline app/views/admin/creator_settings/new.html.erb:40]: Refactor (blocking): No need for any width CSS utility classes or the `flex` CSS utility class. The parent, `.crayons-field` has its display set to flex with a flex-flow of column.

image

```suggestion
        <%= submit_tag "Finish", class: "crayons-btn btn--primary" %>
```

**Review by Ridhwana — APPROVED** [review]
This looks good to me and makes sense but maybe just check with @anujbhavsar96.

**Review by nickytonline — CHANGES_REQUESTED** [review]

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @juliannatetreault! 🚀

**Comment by anujbhavsar96** [comment]
This looks great @juliannatetreault 🚀

I think it'll look better if we keep the width 400/480px. Why this? Because whatever the inputs we are asking are not huge multi-word inputs. So there is no need to have bigger field sizes. 

image

**Pull request merged**